src/arrayMethodReduce.js

```diff
@@ -5,7 +5,14 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let value = arguments.length === 1 ? this[0] : initialValue;
```

Comment:
```suggestion
    let value = initialValue || this[0]
```

Reply:
I check it and it doesn't work because we can put an empty string or 0 in the second argument. Both of them will be converted to boolean and it will be false. For example here we want to get string as a result but we will get a number [1,2,3,4].reduce((acc, e)=>{ return acc+ e }, "") because acc is false so we assign first element as initialValue

Reply:
yes, I agree, so we can make it better by just checking if initialValue != undefined
```suggestion
    let value = initialValue != undefined ? initialValue : this[0]
```

Reply:
there is a case in test where we have undefined as a second argument and we have to use it as initialValue
![image](https://user-images.githubusercontent.com/90928916/216115896-e3ed4941-6966-4ad2-b2e1-95094df028b9.png)

Reply:
Yes, you are right.
checking arguments length looks better)
You do not need to re-request review again, because this task is already approved 🙂